pilot/cmd/pilot-agent/main.go

```diff
@@ -476,18 +479,33 @@ func getDNSDomain(domain string) string {
 	return domain
 }
 
-// check if SDS UDS path and token path exist, if both exist, requests key/cert
-// using SDS instead of secret mount.
-func detectSds(controlPlaneBootstrap, controlPlaneAuthEnabled bool, udspath, trustworthyJWTPath string) (bool, string) {
+// detectSds checks if the SDS address (when it is UDS) and JWT paths are present.
+func detectSds(controlPlaneBootstrap bool, sdsAddress, trustworthyJWTPath string) (bool, string) {
 	if !sdsEnabledVar.Get() {
 		return false, ""
 	}
 
+	if len(sdsAddress) == 0 {
+		return false, ""
+	}
+
+	// sdsAddress will not be empty when sdsAddress is a UDS address.
+	udsPath := ""
+	if strings.HasPrefix(sdsAddress, "unix:") {
+		udsPath = strings.TrimPrefix(sdsAddress, "unix:")
+		if len(udsPath) == 0 {
+			// If sdsAddress is "unix:", it is invalid, return false.
+			return false, ""
+		}
+	}
```

Comment:
else { return true, udsPath}

Reply:
I mean return true, trustworthyPath

Reply:
trustworthyPath is checked below to make sure token is ready.

Reply:
I think we still need to stat the trustworthy JWT path, no? Or are you saying stat the trustworthy JWT then return to reduce the complexity of the code below?

Reply:
@costinm Please take a look.

Reply:
Great !